idk/kafka/cmd_test.go

```diff
@@ -682,3 +702,122 @@ func tDoHTTPPost(t *testing.T, url, contentType, body string) string {
 
 	return string(bod)
 }
+
+/*
+In some cases, when we see max.poll.interval.ms exceeded on the kafka consumer,
+we also see the consumer hang when it's trying to close properly. See SUP-288.
+This test:
+ 1. configures the consumer such that it will hang if max.poll.interval.ms is exceeded.
+ 2. checks that
+    a. the consumer did leave the group due to exceeding max.poll.interval.ms
+    b. we close the consumer and log that we weren't able to confirm it closed properly
+
+It could be possible this test fails even though there isn't necessarily an issue. This
+would be the case if a fix comes along to the kafka-go lib that prevents the consumer
+from hanging during the call it's consumer.Close() func.
+*/
+func TestCloseTimeout(t *testing.T) {
+
+	//define some vars
+	source := "bank"
+	target := "kafka"
+	now := time.Now().UnixNano()
+	index := fmt.Sprintf("close_timeout_%d", now)
+	topic := fmt.Sprintf("close_timeout_%d", now)
+	subject := fmt.Sprintf("close_timeout_%d_subject", now)
+	log_path := fmt.Sprintf("./consumer_%d.log", now)
+	genNumRecords := 5000
+
+	// configure the consumer
+	consumer, err := NewMain()
+	if err != nil {
+		t.Fatalf("issues creating a consumer")
+	}
+	configureTestFlags(consumer)
+	consumer.Topics = []string{topic}
+	consumer.BatchSize = 1000
+	consumer.IDField = "user_id"
+	consumer.KafkaGroupInstanceId = "some_id"
+	consumer.ConsumerCloseTimeout = 3
+	consumer.KafkaMaxPollInterval = "15000"
```

Comment:
I assume this is ... milliseconds as a string? That is certainly a novel data type. :P

I would be inclined to pick a significantly shorter timeout in testing because in theory none of this should take more than a second or two, right?

Reply:
So, I remember playing around with all the timeouts & what not. Dug into it a bit more and here is what I found.

- Session timeouts > 6 seconds give the following error:
```
%7|1678230252.773|REJOIN|rdkafka#consumer-1| [thrd:main]: Group "defaultgroup": Rejoining group without an assignment: JoinGroup error: Broker: Invalid session timeout
```
- There could be a few reasons for this but I haven't dug into it much further.
- max poll interval needs to be > that session timeout so it's at 6.5 seconds.
- the transaction lock needs to be a couple seconds > that the poll interval or else we don't get the timeout we're after so it's set at 9 seconds.
- We have to wait for the transaction lock to end before the close timeout clock starts ticking. Once it doesn't we have to wait for that close timeout to timeout.
- I'm hesitant dropping the close timeout in the case it actually takes some time to close property. Then we could get a false positive (the consumer was going to close properly but we didn't give it enough time.
- So then we have to wait ~15 seconds to know if the consumer is actually hanging (9 sec transaction lock + 3 second close timeout)

So anyways, let me know what you think. Is it worth digging into the session timeout error when < 6 seconds?

Reply:
Hmm. I think... probably not right now? That feels like a thing that could be any of several other timeout issues compounding, and it feels like we don't need to explore it right now. Maybe make a note or a comment somewhere so if someone comes looking they get that context.